File: js/Chess/ChessProblem.test.js
import ChessProblem from './ChessProblem'
import ChessStatus from './ChessStatus'
import fs from 'fs'

describe('ChessProblem', () => {
    test('repeated method call do not change behaviour', () => {
        const boardLines = [
            "        ",
            "        ",
            "        ",
            "   q    ",
            "    K   ",
            " Q      ",
            "        ",
            "        ",
        ];

        testChess(true, ChessStatus.check, boardLines);

        // Now check that internal board modifications during the first call do not change answer
        testChess(false, ChessStatus.check);
    });

    test('all tests', () => {
        const dirPath = './Chess/ChessTests';
        const readFile = (path) => fs.readFileSync(path).toString().trim().split('\n').map(line => line.trim());
        const allFilesNames = fs.readdirSync(dirPath);
        const inputFiles = allFilesNames.filter(fileName => fileName.endsWith('.in'));
        const answerFiles = allFilesNames.filter(fileName => fileName.endsWith('.ans'));

        for (let i = 0; i < inputFiles.length; i++) {
            const boardLines = readFile(dirPath + '/' + inputFiles[i]);
            const expectedAnswer = fs.readFileSync(dirPath + '/' + answerFiles[i]).toString().trim();
            testChess(true, expectedAnswer, boardLines);
        }
    });
});

function testChess (needLoad, expectedAnswer, boardLines) {
    if (needLoad) {
        ChessProblem.loadFrom(boardLines);
    }
    ChessProblem.calculateChessStatus();
    expect(ChessProblem.chessStatus).toBe(expectedAnswer);
}

File: cs/Chess/ChessProblem_Test.cs
using System;
using System.IO;
using NUnit.Framework;

namespace Chess
{
    [TestFixture]
    public class ChessProblem_Test
    {
        [Test]
        public void RepeatedMethodCallDoNotChangeBehaviour()
        {
            var boardLines = new[]
            {
                "        ",
                "        ",
                "        ",
                "   q    ",
                "    K   ",
                " Q      ",
                "        ",
                "        ",
            };
            var board = new BoardParser().ParseBoard(boardLines);
            var status1 = new ChessProblem(board).CalculateChessStatus();
            Assert.AreEqual(ChessStatus.Check, status1);

            // Now check that internal board modifications during the first call do not change answer
            var status2 = new ChessProblem(board).CalculateChessStatus();
            Assert.AreEqual(ChessStatus.Check, status2);
        }

        [Test]
        public void AllTests()
        {
            var dir = TestContext.CurrentContext.TestDirectory;
            var testsCount = 0;
            foreach (var filename in Directory.GetFiles(Path.Combine(dir, "ChessTests"), "*.in"))
            {
                TestOnFile(filename);
                testsCount++;
            }
            Console.WriteLine("Tests passed: " + testsCount);
        }

        private static void TestOnFile(string filename)
        {
            var boardLines = File.ReadAllLines(filename);
            var board = new BoardParser().ParseBoard(boardLines);
            var expectedAnswer = File.ReadAllText(Path.ChangeExtension(filename, ".ans")).Trim();
            var status = new ChessProblem(board).CalculateChessStatus();
            Assert.AreEqual(expectedAnswer, status.ToString().ToLower(), "Failed test " + filename);
        }
    }
}